backend/controllers/chatController.js

```diff
@@ -235,12 +233,104 @@ function stringtify_for_sse(raw_data) {
   return `data: ${JSON.stringify(raw_data)}\n\n`;
 }
 
+// TO join a chat:
+// 1. Validate chat_pin.
+// 2. Repond with exists: true if found
+// 3. Respond with error and exists: false if not.
+async function join_chat(req, res) {
+	console.log("GET request received");
+  const chat_pin = req.params.chat_id;
+  try {
+    console.log("Attempting to find chat");
+    validate_chat_pin(chat_pin);
+    console.log("Chat found");
+    res.status(200).json({ exists: true });
+  } catch (error) {
+    console.error("Error checking chatroom existence:", error);
+    respond_with_error_json(res, 400, {exists: false, message: error.message});
+  }
+}
```

Comment:
For now I may just handle it for guest user, but once we have cookie implmented, we can then handle for users with account. What do you think?

Reply:
Alright, if we didn't wanna do cookies we could also just have the user_id stored inside the HTML

Reply:
But then whichever page we redirect a user to, we will have to send that user_id around which is quite a lot of work for client-side Js. If we use cookie, we can just simply assign user_id for the browser to keep and can even simplify the post message route (since we can just extract the user_id from the cookie, instead of passing the user_name around)